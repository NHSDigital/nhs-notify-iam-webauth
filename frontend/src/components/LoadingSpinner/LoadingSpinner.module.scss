@use 'nhsuk-frontend/packages/core/settings/colours' as colours;

.nhsuk-loader {
  position: relative;
  text-align: center;
  height: 400px;
  padding-top: 130px
}

.nhsuk-loader__spinner {
  animation: spin 2.5s linear infinite;
  border: 10px solid colours.$color_nhsuk-grey-4;
  border-radius: 50%;
  border-top: 10px solid colours.$color_nhsuk-blue;
  display: inline-block;
  height: 120px;
  margin: auto;
  width: 120px;
}

.nhsuk-loader__text {
  display: block;
  margin-bottom: 0;
  font-size: 1.4em;
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}