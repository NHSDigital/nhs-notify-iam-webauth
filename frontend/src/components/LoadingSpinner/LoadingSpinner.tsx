import styles from './LoadingSpinner.module.scss';

export function LoadingSpinner({ text }: { text: string }) {
  return (
    <div className={styles['nhsuk-loader']}>
      <span className={styles['nhsuk-loader__spinner']} />
      <h1 className={styles['nhsuk-loader__text']}>{text}</h1>
    </div>
  );
}
