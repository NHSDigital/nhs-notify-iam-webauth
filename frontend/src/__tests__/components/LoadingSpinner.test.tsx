import { render } from '@testing-library/react';
import LoadingSpinner from '@/components/LoadingSpinner/LoadingSpinner';

describe('LoadingSpinner component', () => {
  it('renders component correctly', async () => {
    const container = render(<LoadingSpinner text='loading text' />);

    expect(container.asFragment()).toMatchSnapshot();
  });
});
