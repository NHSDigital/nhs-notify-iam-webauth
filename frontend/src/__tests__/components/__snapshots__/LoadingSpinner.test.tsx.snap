// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner component renders component correctly 1`] = `
<DocumentFragment>
  <div
    class="nhsuk-loader"
  >
    <span
      class="nhsuk-loader__spinner"
    />
    <h1
      class="nhsuk-loader__text"
    >
      loading text
    </h1>
  </div>
</DocumentFragment>
`;
